Gems/RecastNavigation/Code/Source/Misc/RecastHelpers.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+#include <DetourAlloc.h>
+#include <Recast.h>
+#include <AzCore/Math/Aabb.h>
+#include <AzCore/Math/Vector3.h>
+
+namespace RecastNavigation
+{
+    //! A helper class to manage different coordinate systems between O3DE and Recast Navigation library.
+    //! In O3DE, +Z is up. In Recast library, +Y is up.
+    class RecastVector3
+    {
+    public:
+        RecastVector3() = default;
+
+        //! A constructor from O3DE coordinate values.
+        explicit RecastVector3(const AZ::Vector3& in)
+        {
+            m_x = in.GetX();
+            m_y = in.GetZ(); // swapping y and z
+            m_z = in.GetY();
+        }
+
+        //! A constructor from Recast coordinate values.
+        explicit RecastVector3(const float* data)
+        {
+            m_x = data[0];
+            m_y = data[1];
+            m_z = data[2];
+        }
+
+        //! @returns raw data without any conversion between coordinate systems.
+        float* GetData() { return &m_x; }
```

Comment:
comment: I do think this is technically UB 😝 I know it _technically_ works but doing a pointer addition between members is not really allowed... it might be worth thinking of an alternate approach (maybe internally we should just store a `float xyz[3];` array with accessors

Reply:
Oh, yeah, I like that. Switching to m_xyz[3].

Reply:
Awesome thanks! 😄